### PR TITLE
S3 Improvements

### PR DESCRIPTION
**Pull request by flovilmart**
Proposed fix for #731 and #628 

- Adds generic testing for Files Adapters
- Adds support for content type setting when uploading to S3.
- Adds ability to enable S3 adapter tests with environment variables S3_ACCESS_KEY, S3_SECRET_KEY

### CODE REVIEW

**Comment by drew-gross** [comment]
Looks good! Can we move towards making this an external repo? Then `FileAdapter` implementers can run the tests without depending on Parse Server

**Comment by flovilmart** [comment]
I wrote the test to make it easier to move to another repo. We should merge that, first, then move all adapters to 3rd party repos.

@gfosco had some thoughts on that too

File: src/Controllers/FilesController.js [hunk 1]
```diff
@@ -10,10 +12,22 @@ export class FilesController extends AdaptableController {
...
+      filename = filename + '.' + mime.extension(contentType);
+    } else if (hasExtension && !contentType) {
+      contentType = mime.lookup(filename);
+    }
+
     filename = randomHexString(32) + '_' + filename;
+
     var location = this.adapter.getFileLocation(config, filename);
-    return this.adapter.createFile(config, filename, data).then(() => {
+    return this.adapter.createFile(config, filename, data, contentType).then(() => {
```
Comment (nlutsenko) [inline src/Controllers/FilesController.js]: Since this is sort of breaking - can we add this to the docs on FilesAdapter?

Comment (flovilmart) [inline src/Controllers/FilesController.js]: yeppa

**Pull request merged**